control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java

```diff
@@ -0,0 +1,207 @@
+        try {
+            HttpResponse response = httpClient.execute(request);
+            HttpEntity entity = response.getEntity();
+            String responseString = EntityUtils.toString(entity, "UTF-8");
+            if (responseString == null) {
+                throw new ActionParamsException("Didn't get any response from service");
+            }
+            return responseString;
+        } catch (IOException ex) {
+            throw new ActionException("Error posting the WFS-T message to service", ex);
+        }
+    }
+
+    protected boolean isAllowedGeomType(String type) {
+        if (type == null) {
+            return false;
+        }
+        return ALLOWED_GEOM_TYPES.contains(type);
+    }
+
+    protected String getGeometry(String geometryType, JSONArray data, String srsName) throws ActionParamsException, JSONException {
+        if ("multipoint".equals(geometryType)) {
+            return getMultipoint(srsName, data);
+        } else if ("multilinestring".equals(geometryType)) {
+            return getMultiLineStringGeometries(srsName, data);
+        } else if ("multipolygon".equals(geometryType)) {
+            return getMultiPolygonGeometries(srsName, data);
+        }
+        throw new ActionParamsException("Unknown type");
+    }
+
+    protected String getMultipoint(String srsName, JSONArray data) throws JSONException {
+        StringWriter writer = new StringWriter();
+        writer.append("<gml:MultiPoint xmlns:gml=\"http://www.opengis.net/gml\" srsName=\"");
+        writer.append(srsName);
+        writer.append("\">");
+        for (int i = 0; i < data.length(); i++) {
+            writer.append("<gml:pointMember><gml:Point><gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                    + data.getJSONObject(i).getString("x") + "," + data.getJSONObject(i).getString("y")
+                    + "</gml:coordinates></gml:Point></gml:pointMember>");
+        }
+        writer.append("</gml:MultiPoint>");
+        return writer.toString();
+    }
+
+    protected String getMultiLineStringGeometries(String srsName, JSONArray data) throws JSONException {
+        StringWriter writer = new StringWriter();
+        writer.append("<gml:MultiLineString xmlns:gml=\"http://www.opengis.net/gml\" srsName=\"");
+        writer.append(srsName);
+        writer.append("\">");
+        for (int lineIndex = 0; lineIndex < data.length(); lineIndex++) {
+            writer.append("<gml:lineStringMember><gml:LineString><gml:coordinates decimal=\".\" cs=\",\" ts=\" \">");
+            JSONArray lineCoordinates = data.getJSONArray(lineIndex);
+            for (int coordinateIndex = 0; coordinateIndex < lineCoordinates.length(); coordinateIndex++) {
+                JSONObject coord = lineCoordinates.getJSONObject(coordinateIndex);
+                writer.append(coord.getString("x") + "," + coord.getString("y"));
+                if (coordinateIndex < (lineCoordinates.length() - 1)) {
```

Comment:
Or move to the start of the loop and check `coordinateIndex > 0`. But don't bother changing it, there's too much to fix...